template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kafka/KafkaVolumeConfigProvider.java

```diff
@@ -19,10 +20,12 @@
 
     @Override
     public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, HostgroupView hostGroupView, TemplatePreparationObject source) {
-        Integer volumeCount = Objects.nonNull(hostGroupView) ? hostGroupView.getVolumeCount() : 0;
+        Optional<HostgroupView> hostGroup = source.getHostGroupsWithComponent(KafkaRoles.KAFKA_BROKER)
+                        .min(Comparator.comparing(HostgroupView::getVolumeCount));
+        Integer minimumVolumeCount = hostGroup.isPresent() ? hostGroup.get().getVolumeCount() : 0;
         switch (roleType) {
             case KafkaRoles.KAFKA_BROKER:
-                return List.of(config("log.dirs", buildVolumePathStringZeroVolumeHandled(volumeCount, "kafka")));
+                return List.of(config("log.dirs", buildVolumePathStringZeroVolumeHandled(minimumVolumeCount, "kafka")));
```

Comment:
this will be applied on datalakes as well. do we comfortable with this?

Reply:
Added a condition to only apply the minimum count for workload stacks.